lib/i18n_getstrings.dart

```diff
@@ -2,24 +2,63 @@ import 'dart:io';
 
 import 'package:analyzer/dart/analysis/utilities.dart';
 import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
 import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:equatable/equatable.dart';
+
+enum I18nRequiredModifiers { plural }
+
+class I18nSuffixes {
+  static const String i18n = 'i18n';
+  static const String fill = 'fill';
+  static const String plural = 'plural';
+  static const String version = 'version';
+  static const String allVersions = 'allVersions';
+
+  static const List<String> allSuffixes = const [
+    i18n,
+    fill,
+    plural,
+    version,
+    allVersions,
+  ];
+}
+
+class ExtractedString extends Equatable {
+  final String string;
+  final bool pluralRequired;
+
+  ExtractedString(this.string, {this.pluralRequired = false});
+
+  @override
+  List<Object> get props => [string, pluralRequired];
+}
+
+class DecodedSintax {
+  DecodedSintax._({
+    required this.valid,
+    this.modifiers = const [],
+  });
+
+  DecodedSintax.valid(List<I18nRequiredModifiers> modifiers)
+      : this._(valid: true, modifiers: modifiers);
+
+  DecodedSintax.invalid() : this._(valid: false);
+
+  final bool valid;
+  final List<I18nRequiredModifiers> modifiers;
```

Comment:
This looks a bit complicated for what you are trying to achieve. Wouldn't it be better to just add a `bool isPlural` to the class?